src/Simplify.cpp

```diff
@@ -4303,60 +4362,99 @@ class Simplify : public IRMutator {
                 }
             }
 
-            // Try to collapse an interleave of strided loads of ramps
-            // from the same buffer into a single load of a ramp.
-            if (const Load *first_load = new_vectors[0].as<Load>()) {
-                vector<Expr> load_predicates;
-                vector<Expr> load_indices;
-                bool always_true = true;
-                for (Expr e : new_vectors) {
-                    const Load *load = e.as<Load>();
-                    if (load && load->name == first_load->name) {
-                        load_predicates.push_back(load->predicate);
-                        load_indices.push_back(load->index);
-                        always_true = always_true && is_one(load->predicate);
+            // Try to collapse an interleave of slices of vectors from
```

Comment:
Thank you for this. This is excellent. When I was first thinking of implementing vrmpy and vdmpy I had been discouraged by having to check all of this in HexagonOptimize.cpp/CodeGen_Hexagon.cpp. But, this is definitely better that it is simplified in Simplify.cpp.